### PR TITLE
Adding the T-shirt shop to the Java codebase to

### PR DESCRIPTION
**Pull request by JlUgia**
This is a preliminary effort to bring the Java application up to speed with the Kotlin version.
This PR adds a replica of the [SHOP demo](http://g.co/pay/demo) on the Web to make it look more like an actual marketplace.

Please do look for code style consistency, good practices, structure and readability.

### CODE REVIEW

File: java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java [hunk 1]
```diff
@@ -192,53 +222,44 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
-              .getString("token")
-              .equals("examplePaymentMethodToken")) {
-        AlertDialog alertDialog =
-            new AlertDialog.Builder(this)
+
+      final JSONObject tokenizationData = paymentMethodData.getJSONObject("tokenizationData");
+      final String tokenizationType = tokenizationData.getString("type");
+      final String token = tokenizationData.getString("token");
+
+      if ("PAYMENT_GATEWAY".equals(tokenizationType) && "examplePaymentMethodToken".equals(token)) {
```
Comment (stephenmcd) [inline java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java:243]: I'm not sure if there's a convention for this, but they look backwards to me, eg:

`if (tokenizationType.equals("PAYMENT_GATEWAY") && token.equals("examplePaymentMethodToken")) {`

File: java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java [hunk 2]
```diff
@@ -192,53 +222,44 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
   private void handlePaymentSuccess(PaymentData paymentData) {
-    String paymentInformation = paymentData.toJson();
 
     // Token will be null if PaymentDataRequest was not constructed using fromJson(String).
-    if (paymentInformation == null) {
-      return;
-    }
-    JSONObject paymentMethodData;
+    final String paymentInfo = paymentData.toJson();
+    if (paymentInfo == null) return;
```
Comment (stephenmcd) [inline java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java:232]: Not sure what the convention is, but further down the file (line 292) I see a 1-line if statement with a return that includes brackets, and this one doesn't, so we should use one style or the other. My preference would be to include brackets, eg: 

```
if (paymentInfo == null) {
    return;
}
```

File: java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java (hunk 1 above)
Comment (JlUgia) [reply java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java:243]: This prevents a `NullPointerException` f any of `tokenizationType` or `token` are `null`. Putting the definite value first makes the operation work resiliently no matter what's on the other side (eg.: if `tokenizationType` is `null`, the condition will test `false`)

File: java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java (hunk 2 above)
Comment (JlUgia) [reply java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java:232]: Adding

File: java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java (hunk 1 above)
Comment (stephenmcd) [reply java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java:245]: That's clever and makes sense, thanks

**Pull request merged**